### ISSUE TITLE
wsltty does not start anymore with recent wsl2 / Win11

### ISSUE THREAD

**Issue by g2flyer** [mintty/wsltty#354]
Window quickly shows up and quickly disappears; when starting with `-h always`, it shows `/bin/wslbridge2: Exit 1`.  I guess again a change in windows not-so-documented APIs similar to issues #342 and #343 which makes wslbridge2 fail?  Can't point finger at any particular KB install though, probably KB5039895 or KB5040442?

**Comment by appomsk** [comment]
Some kind of miracles. It stopped working, I rebooted the computer - it works. Now I discovered that it doesn't work again.

**Comment by g2flyer** [comment]
> re

Definitely a reboot does _not_ help in my case ...

**Comment by McG800** [comment]
This has been an issue for me for a while as well. @g2flyer Do you still have the same issue?

**Comment by g2flyer** [comment]
> This has been an issue for me for a while as well. @g2flyer Do you still have the same issue?

Yep, still an issue with WSL & windows versions as follows
 ```
> wsl --version
WSL version: 2.3.17.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.64
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4169
```

**Comment by McG800** [comment]
I'm on similar versions as well.

> WSL version: 2.3.17.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.64
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22635.4145

**Comment by appomsk** [comment]
I have Windows 11 (professional). Mintty stopped working some time ago, but it was possible to run Linux (Ubuntu) through the Windows terminal. After the September Windows update, Linux stopped running through the terminal as well. It was giving some sort of wsl error. After a google search and various trials, I installed Hyper-V, which I had disabled. Now everything is working.

**Comment by mintty** [comment]
> I installed Hyper-V, which I had disabled. Now everything is working.

So wsltty works again too, right? Can others confirm that please?
Also please provide a link how to download/install Hyper-V.

**Comment by McG800** [comment]
I believe Hyper-V is installed when you enable it in the "Turn Windows Features on or Off" app in Windows 11. Which mine is and has been.

**Comment by appomsk** [comment]
Unfortunately, I wasn't paying much attention to my actions. I was actually ready to tear down Windows and install Linux.

Which I can definitely attest to:

1. I uninstalled Hyper-V some time ago (because I thought some of the problems with VirtualBox were because of it. It wasn't, as it turns out).

2. wsltty stopped working completely this summer. But wsl was working through Windows Terminal.

3. In September, it stopped working as well. The error was something like - wsl component not installed.

4. I googled this error, found some answers, some of them were wrong, but by trial and error I came up with the Hyper-V installation (it's in Windows components in the Applications menu in Settings). After installing and rebooting, I found that everything worked.

5. Theoretically, I did some other actions, according to all sorts of answers found on MSDN mostly - enabled services, reinstalled wsl-components, restarted them. So there's a certain possibility that it was just coincidental and not directly related to Hyper-V.

6. Anyway, after “dancing with tambourine” everything worked.

I apologize again for not logging my actions, but I say I was a little angry and was ready to tear down this Windows, from which with every update constantly come some problems.
